keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,363 @@
+support some Kubernetes functionality like e.g. Services, I am willing to give
+that up for performance.
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster operator I wish to isolate workloads based on
+namespaces and network access by overriding the Cluster Default PodNetwork for
+a given Namespace. I do not want the tenants to change their manifests for that
+purpose. Those workloads should have the same level of Kubernetes functionality:
+Services, NetworkPolicies, access to Kubernetes API. I wish to support
+“profiles” for Namespace, where I can not only change Cluster Default PodNetwork,
+but define a set of PodNetworks assigned to given Namespace that Pods created in
+that Namespace are automatically attached to.
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I need the ability to add/remove Network Interfaces to
+existing VMs without re-creating the VM. This would not be applicable for the
+Primary PodNetwork.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary PodNetwork.
```

Comment:
you can support SLAAC today nothing stops pods to get the IPs from slaac ... other thing is reporting that to the Pod.Status, but this is based on how kubelet-> Containter-runtimes-> CNI works today, but nothing is stopping anybody to use slaac and report back those ips 

The other thing is the pod lifecycle, the pod lifecycle and its phases https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase makes that the network is not updated since it is created, but the pod.Status.PodIP is mutable, nothing stops anybody to update but ... here by dragons (and you need enough permissions)

Reply:
This is about the other half, i.e. Hot-Plug IP. Additionally as mentioned at the top of this section:

> References to technologies or exact products does not
indicate that this API will directly support them.